atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -60,34 +60,38 @@ public KeyValueService createRawKeyValueService(
             LongSupplier freshTimestampSource,
             boolean initializeAsync) {
         AtlasDbVersion.ensureVersionReported();
-        Refreshable<CassandraKeyValueServiceRuntimeConfig> cassandraRuntimeConfig =
-                preprocessKvsRuntimeConfig(runtimeConfig);
+        CassandraKeyValueServiceConfig configWithNamespace = getConfigWithNamespace(config, namespace);
+        // Safety: CassandraReloadableKVSRuntimeConfig is a subtype of CassandraKVSRuntimeConfig, but Refreshable isn't
+        // covariant in the generic type arg, so the cast is required.
+        Refreshable<CassandraKeyValueServiceRuntimeConfig> mergedRuntimeConfig = createMergedKeyValueServiceConfig(
+                        configWithNamespace, preprocessKvsRuntimeConfig(runtimeConfig))
+                .map(CassandraKeyValueServiceRuntimeConfig.class::cast);
```

Comment:
Cast is here instead of the createMerged method just because createDerived actually wants the reloadable kvs, not the base config